storage_scrubber/src/checks.rs

```diff
@@ -114,16 +126,40 @@ pub(crate) fn branch_cleanup_and_check_errors(
                         }
 
                         if !tenant_objects.check_ref(id.timeline_id, &layer, &metadata) {
-                            // FIXME: this will emit false positives if an index was
-                            // uploaded concurrently with our scan.  To make this check
-                            // correct, we need to try sending a HEAD request for the
-                            // layer we think is missing.
-                            result.errors.push(format!(
-                                "index_part.json contains a layer {}{} (shard {}) that is not present in remote storage",
+                            let timeline_root = root_target.timeline_root(id);
+                            let remote_layer_path = format!(
+                                "{}{}{}",
+                                timeline_root.prefix_in_bucket,
                                 layer,
-                                metadata.generation.get_suffix(),
-                                metadata.shard
-                            ))
+                                metadata.generation.get_suffix()
```

Comment:
Unsure if this covers all cases, like we have older timelines without the suffix, some with `-v1-generation` and some simply `-generation`.

Reply:
I will double check the `-v1-<generation>` case. The other case seems to be handled by the `Display` impl.

https://github.com/neondatabase/neon/blob/3773a945f517a2d451397cc25bd887a710ef8574/libs/utils/src/generation.rs#L94-L102

Reply:
end up using the `remote_layer_path` function.